src/spell/modules/spell/spell/src/mailbox.rs

```diff
@@ -0,0 +1,220 @@
+use marine_rs_sdk::marine;
+
+use fluence_spell_dtos::value::{StringValue, UnitValue};
+
+use crate::auth::is_by_spell;
+use crate::kv::primitive::read_string;
+use crate::misc::fetch_rows;
+use crate::schema::db;
+
+#[marine]
+/// `messages` contains up to `DEFAULT_MAX_MAILBOX` latest messages,
+/// sorted in the order they were pushed
+pub struct GetMailboxResult {
+    pub messages: Vec<String>,
+    pub success: bool,
+    pub error: String,
+}
+
+impl From<eyre::Result<Vec<String>>> for GetMailboxResult {
+    fn from(result: eyre::Result<Vec<String>>) -> Self {
+        match result {
+            Ok(messages) => GetMailboxResult {
+                messages,
+                success: true,
+                error: "".to_string(),
+            },
+            Err(e) => GetMailboxResult {
+                success: false,
+                error: format!("get_mailbox error: {}", e),
+                messages: vec![],
+            },
+        }
+    }
+}
+
+#[marine]
+/// Push a message to the mailbox. Mailbox keeps `DEFAULT_MAX_MAILBOX` latest messages.
+pub fn push_mailbox(message: String) -> UnitValue {
+    let call_parameters = marine_rs_sdk::get_call_parameters();
+
+    // We want to prevent anyone except this spell to store logs to its kv
+    if !is_by_spell(&call_parameters) {
```

Comment:
I guess the point of the mailbox is to communicate with other spells and peers, so we should allow them to write there

Reply:
yeah, i think you're right